### PR TITLE
[DOCS] Add get snapshot repo API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Adds API reference docs for the get snapshot repo API.

Relates to #56069

### CODE REVIEW

File: docs/reference/snapshot-restore/apis/get-repo-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+(Optional, boolean) If `true`, the request gets information from the local node
+only. If `false`, the request gets information from the master node. Defaults to
+`false`.
+
+`master_timeout`::
+(Optional, <<time-units, time units>>) Specifies the period of time to wait for
+a connection to the master node. If no response is received before the timeout
+expires, the request fails and returns an error. Defaults to `30s`.
+
+`timeout`::
```
Comment (original-brownbear) [inline docs/reference/snapshot-restore/apis/get-repo-api.asciidoc:60]: I don't think we're using this parameter right now, let's not document it here

**Review by original-brownbear — APPROVED** [review]
LGTM thanks @jrodewig just one thing to remove :)

File: docs/reference/snapshot-restore/apis/get-repo-api.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/snapshot-restore/apis/get-repo-api.asciidoc:60]: Thanks for catching this.

**Comment by jrodewig** [comment]
### Backport commits
master 96a8ba4
7.x f4a673c
7.8 cc8126d 
7.7 b0ab9d9

**Pull request merged**